File: app-core/src/main/java/kr/flab/wiki/core/login/LoginUtils.java
package kr.flab.wiki.core.login;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class LoginUtils {

    private LoginUtils(){}

    public static boolean isValidEmail(String email){

        String regex = "^[_a-z0-9-]+(.[_a-z0-9-]+)*@(?:\\w+\\.)+\\w+$";
        Pattern pattern = Pattern.compile(regex);
        Matcher matcher = pattern.matcher(email);

        if(matcher.matches()){
            return true;
        }
        return false;
    }
}

File: app-core/src/main/kotlin/kr/flab/wiki/core/login/LoginUtils.kt
package kr.flab.wiki.core.login

import java.util.regex.Pattern

object LoginUtils {
    // https://atin.tistory.com/425
    private const val REGEX_EMAIL = "^[_a-z0-9-]+(.[_a-z0-9-]+)*@(?:\\w+\\.)+\\w+$"
    private val emailPattern: Pattern = Pattern.compile(REGEX_EMAIL)

    fun isValidEmail(email: String?): Boolean {
        return emailPattern.matcher(email).matches()
    }
}
